### PR TITLE
Application warning banner

### PR DESCRIPTION
**Pull request by mpmadhavig**
### Purpose
This PR will add a banner to the Application edit to maintain the below configs.

1. useClientIdAsSubClaimForAppTokens
2. omitUsernameInIntrospectionRespForAppTokens

The initial UI design is as follows.

https://github.com/user-attachments/assets/092abdbe-66a5-4813-98d4-1bdb359c6b4c

### Related Issues
- https://github.com/wso2/product-is/issues/20917

### Related PRs
- https://github.com/wso2/identity-apps/pull/6783

### Checklist

- [x] e2e cypress tests locally verified. (for internal contributers)
- [x] Manual test round performed and verified.
- [x] UX/UI review done on the final implementation.
- [x] Documentation provided. (Add links if there are any)
- [ ] Relevant backend changes deployed and verified
- [x] [Unit tests](/docs/testing/UNIT_TESTING.md) provided. (Add links if there are any)
- [x] [Integration tests](https://github.com/wso2/product-is/tree/master/modules/integration) provided. (Add links if there are any)

### Security checks
- [x] Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines
- [x] Ran FindSecurityBugs plugin and verified report
- [x] Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets

### CODE REVIEW

**Comment by Achintha444** [comment]
### Design comments

1. IMO it is not required the box title in the view more section.
2. IMO, it is not required to have the numbered list, only the check box is enough.
3. Content in the box, including the main button should be align with the content of the alert box.

File: features/admin.applications.v1/pages/application-edit.tsx [hunk 1]
```diff
@@ -57,13 +67,17 @@ import {
...
     ApplicationTemplateListItemInterface,
+    OIDCDataInterface,
     State,
     SupportedAuthProtocolTypes,
     idpInfoTypeInterface
 } from "../models";
 import { ApplicationManagementUtils } from "../utils/application-management-utils";
 import { ApplicationTemplateManagementUtils } from "../utils/application-template-management-utils";
 import "./application-edit.scss";
+import { Typography } from "@mui/material";
```
Comment (Achintha444) [inline features/admin.applications.v1/pages/application-edit.tsx:78]: do not directly import from material, use `https://wso2.github.io/oxygen-ui/react/?path=/docs/data-display-typography--overview` instead.

File: features/admin.applications.v1/pages/application-edit.tsx [hunk 2]
```diff
@@ -129,6 +143,44 @@ const ApplicationEditPage: FunctionComponent<ApplicationEditPageInterface> = (
...
+        if (!isBannerDataLoading) {
+            setApplicationInboundConfig(applicationInboundConfigData);
+        }
+    }, [ applicationInboundConfigData, isBannerDataLoading ]);
+
+    /**
+     * Assign loaded banner data into config states.
+     */
+    useEffect(() => {
+        if (applicationInboundConfig != undefined) {
```
Comment (Achintha444) [inline features/admin.applications.v1/pages/application-edit.tsx:175]:
```suggestion
        if (applicationInboundConfig) {
```

File: features/admin.applications.v1/pages/application-edit.tsx [hunk 3]
```diff
@@ -506,6 +558,354 @@ const ApplicationEditPage: FunctionComponent<ApplicationEditPageInterface> = (
...
     };
 
+    /**
+     * Resolves the application banner content.
+     *
+     * @returns Alert banner.
+     */
+    const resolveAlertBanner = (): ReactElement => {
+
+        const classes: any = classNames( { "application-outdated-alert-expanded-view": viewBannerDetails } );
```
Comment (Achintha444) [inline features/admin.applications.v1/pages/application-edit.tsx:568]: refrain from using any.

File: features/admin.applications.v1/pages/application-edit.tsx [hunk 4]
```diff
@@ -506,6 +558,354 @@ const ApplicationEditPage: FunctionComponent<ApplicationEditPageInterface> = (
         return null;
     };
 
+    /**
+     * Resolves the application banner content.
+     *
+     * @returns Alert banner.
+     */
+    const resolveAlertBanner = (): ReactElement => {
+
```
Comment (Achintha444) [inline features/admin.applications.v1/pages/application-edit.tsx:567]:
```suggestion
```

File: features/admin.applications.v1/pages/application-edit.tsx [hunk 5]
```diff
@@ -506,6 +558,354 @@ const ApplicationEditPage: FunctionComponent<ApplicationEditPageInterface> = (
...
+    };
+
+    /**
+     * Resolves the application banner view details section.
+     *
+     * @returns Alert banner details.
+     */
+    const resolveBannerViewDetails = (): ReactElement => {
+
+        return (
```
Comment (Achintha444) [inline features/admin.applications.v1/pages/application-edit.tsx:631]:
```suggestion
    const resolveBannerViewDetails = (): ReactElement => {
```

File: features/admin.applications.v1/pages/application-edit.tsx [hunk 6]
```diff
@@ -33,21 +37,27 @@ import { ExtensionTemplateListInterface } from "@wso2is/admin.template-core.v1/m
 import { isFeatureEnabled } from "@wso2is/core/helpers";
 import { AlertLevels, IdentifiableComponentInterface } from "@wso2is/core/models";
 import { addAlert } from "@wso2is/core/store";
+import { Field, Forms } from "@wso2is/forms";
```
Comment (Achintha444) [inline features/admin.applications.v1/pages/application-edit.tsx:40]: do not use `Forms` use `FinalForm` instead.
view this PR for more details, "https://github.com/wso2/identity-apps/pull/6722/files#diff-d94522cb3bcc2277d7af7c0f1c38e4e8d1d7659edb195199b995d045be70b235"

File: features/admin.applications.v1/pages/application-edit.tsx [hunk 7]
```diff
@@ -506,6 +558,354 @@ const ApplicationEditPage: FunctionComponent<ApplicationEditPageInterface> = (
...
+    };
+
+    /**
+     * Resolves the update confirmation modal.
+     *
+     * @returns Confirmation modal.
+     */
+    const confirmationModal = () => {
+
+        return (
```
Comment (Achintha444) [inline features/admin.applications.v1/pages/application-edit.tsx:765]:
```suggestion
    const confirmationModal = (): ReactElement => {
```

File: features/admin.applications.v1/pages/application-edit.tsx [hunk 8]
```diff
@@ -506,6 +558,354 @@ const ApplicationEditPage: FunctionComponent<ApplicationEditPageInterface> = (
...
+                </ConfirmationModal.Content>
+            </ConfirmationModal>
+        );
+    };
+
+    /**
+     * Handles banner content update action which prepares data.
+     */
+    const handleBannerCheckBoxUpdate = (formDataValues: any) => {
+
```
Comment (Achintha444) [inline features/admin.applications.v1/pages/application-edit.tsx:829]:
```suggestion
```

File: features/admin.applications.v1/pages/application-edit.tsx [hunk 9]
```diff
@@ -506,6 +558,354 @@ const ApplicationEditPage: FunctionComponent<ApplicationEditPageInterface> = (
...
+                    </ol>
+                </ConfirmationModal.Content>
+            </ConfirmationModal>
+        );
+    };
+
+    /**
+     * Handles banner content update action which prepares data.
+     */
+    const handleBannerCheckBoxUpdate = (formDataValues: any) => {
```
Comment (Achintha444) [inline features/admin.applications.v1/pages/application-edit.tsx:828]: refrain from using `any`

File: features/admin.applications.v1/pages/application-edit.tsx [hunk 10]
```diff
@@ -506,6 +558,354 @@ const ApplicationEditPage: FunctionComponent<ApplicationEditPageInterface> = (
...
+
+        // Todo: check if we need to send all configs. Though the method is PUT, it works as a PATCH.
+        let values: any = { ...applicationInboundConfig };
```
Comment (Achintha444) [inline features/admin.applications.v1/pages/application-edit.tsx:831]: refrain from using `any`

File: features/admin.applications.v1/pages/application-edit.tsx [hunk 11]
```diff
@@ -506,6 +558,354 @@ const ApplicationEditPage: FunctionComponent<ApplicationEditPageInterface> = (
...
+
+        setFormdata({ ...values });
+        setShowConfirmationModal(true);
+    };
+
+    /**
+     * Handles the banner data update action.
+     */
+    const handleBannerCheckBoxUpdateConfirmation = async (): Promise<void> => {
+
```
Comment (Achintha444) [inline features/admin.applications.v1/pages/application-edit.tsx:867]:
```suggestion
```

**Comment by Achintha444** [comment]
fix eslint issues.

**Comment by mpmadhavig** [comment]
### Updated UI

1. Application edit section

Screenshot 2024-08-26 at 23 37 25

2. View Details view

Screenshot 2024-08-26 at 23 37 37

3. Application update confirmation modal action

Screenshot 2024-08-26 at 23 37 52

File: features/admin.applications.v1/pages/application-edit.tsx (hunk 1 above)
Comment (mpmadhavig) [reply features/admin.applications.v1/pages/application-edit.tsx]: Fixed with 025df58bd7452704342e396767d22883776319ca

File: features/admin.applications.v1/pages/application-edit.tsx (hunk 2 above)
Comment (mpmadhavig) [reply features/admin.applications.v1/pages/application-edit.tsx:180]: Fixed with 025df58bd7452704342e396767d22883776319ca

File: features/admin.applications.v1/pages/application-edit.tsx (hunk 7 above)
Comment (mpmadhavig) [reply features/admin.applications.v1/pages/application-edit.tsx]: Fixed with 025df58bd7452704342e396767d22883776319ca

File: features/admin.applications.v1/pages/application-edit.tsx (hunk 8 above)
Comment (mpmadhavig) [reply features/admin.applications.v1/pages/application-edit.tsx]: Fixed with 025df58bd7452704342e396767d22883776319ca

File: features/admin.applications.v1/pages/application-edit.tsx (hunk 11 above)
Comment (mpmadhavig) [reply features/admin.applications.v1/pages/application-edit.tsx]: Fixed with 025df58bd7452704342e396767d22883776319ca

**Comment by mpmadhavig** [comment]
> ### Design comments
> 
>     1. IMO it is not required the box title in the view more section.
> 
Removed with 007118f157fb1adbc5fd9ac54c862d7182289f42

>     2. IMO, it is not required to have the numbered list, only the check box is enough.
Removed with 007118f157fb1adbc5fd9ac54c862d7182289f42

>     3. Content in the box, including the main button should be align with the content of the alert box.
Fixed with 025df58bd7452704342e396767d22883776319ca

**Comment by Achintha444** [comment]
> > ### Design comments
> > ```
> > 1. IMO it is not required the box title in the view more section.
> > ```
> 
> Removed with [007118f](https://github.com/wso2/identity-apps/commit/007118f157fb1adbc5fd9ac54c862d7182289f42)
> 
> > ```
> > 2. IMO, it is not required to have the numbered list, only the check box is enough.
> > ```
> 
> Removed with [007118f](https://github.com/wso2/identity-apps/commit/007118f157fb1adbc5fd9ac54c862d7182289f42)
> 
> > ```
> > 3. Content in the box, including the main button should be align with the content of the alert box.
> > ```
> 
> Fixed with [025df58](https://github.com/wso2/identity-apps/commit/025df58bd7452704342e396767d22883776319ca)
> 
> Screenshot 2024-08-27 at 21 15 13

### Design comments - Cont.

1. IMO the title above the each checkbox is not necessary, I think we are only rephrasing the same thing in two places. I think we can add a title to the box, which can summarise the both check boxes if necessary.

cc: @savindi7 @jeradrutnam WDYT ?

File: features/admin.applications.v1/pages/application-edit.tsx (hunk 3 above)
Comment (Achintha444) [reply features/admin.applications.v1/pages/application-edit.tsx:571]: I think we can replace `any` with `string` here.

File: features/admin.applications.v1/pages/application-edit.tsx [hunk 12]
```diff
@@ -506,6 +562,355 @@ const ApplicationEditPage: FunctionComponent<ApplicationEditPageInterface> = (
...
+        );
+    };
+
+    /**
+     * Resolves the application banner view details section.
+     *
+     * @returns Alert banner details.
+     */
+    const resolveBannerViewDetails = (): ReactElement => {
+        return (
```
Comment (Achintha444) [inline features/admin.applications.v1/pages/application-edit.tsx:628]:
```suggestion
    const resolveBannerViewDetails = (): ReactElement => {
```

File: features/admin.applications.v1/pages/application-edit.tsx [hunk 13]
```diff
@@ -506,6 +562,355 @@ const ApplicationEditPage: FunctionComponent<ApplicationEditPageInterface> = (
...
+        );
+    };
+
+    /**
+     * Resolves the update confirmation modal.
+     *
+     * @returns Confirmation modal.
+     */
+    const confirmationModal = () => {
+        return (
```
Comment (Achintha444) [inline features/admin.applications.v1/pages/application-edit.tsx:770]:
```suggestion
    const confirmationModal = (): ReactElement => {
```

File: features/admin.applications.v1/pages/application-edit.tsx (hunk 9 above)
Comment (Achintha444) [reply features/admin.applications.v1/pages/application-edit.tsx:833]: add return type

File: features/admin.applications.v1/pages/application-edit.tsx [hunk 14]
```diff
@@ -506,6 +562,355 @@ const ApplicationEditPage: FunctionComponent<ApplicationEditPageInterface> = (
...
+        });
+    };
+
+    /**
+     * Handles the banner data update action.
+     */
+    const handleBannerCheckBoxUpdateConfirmation = async (): Promise<void> => {
+        setBannerUpdateLoading(true);
+
+        return updateAuthProtocolConfig<OIDCDataInterface>(application?.id, formData, SupportedAuthProtocolTypes.OIDC)
```
Comment (Achintha444) [inline features/admin.applications.v1/pages/application-edit.tsx:875]: why returning this function and the return type set as `void` ?

File: features/admin.applications.v1/pages/application-edit.tsx (hunk 3 above)
Comment (mpmadhavig) [reply features/admin.applications.v1/pages/application-edit.tsx:559]: ce3f7f0210aea52dca4e1340286cc5ffcb701a43

File: features/admin.applications.v1/pages/application-edit.tsx (hunk 9 above)
Comment (mpmadhavig) [reply features/admin.applications.v1/pages/application-edit.tsx]: ce3f7f0210aea52dca4e1340286cc5ffcb701a43

File: features/admin.applications.v1/pages/application-edit.tsx (hunk 12 above)
Comment (mpmadhavig) [reply features/admin.applications.v1/pages/application-edit.tsx]: ce3f7f0210aea52dca4e1340286cc5ffcb701a43

File: features/admin.applications.v1/pages/application-edit.tsx (hunk 14 above)
Comment (mpmadhavig) [reply features/admin.applications.v1/pages/application-edit.tsx]: ce3f7f0210aea52dca4e1340286cc5ffcb701a43

**Comment by mpmadhavig** [comment]
### Updated UI designs

1. Application List View
Screenshot 2024-09-27 at 18 20 41

2. One outdated application edit view
Screenshot 2024-09-27 at 18 20 34

3. Expanded banner view
Screenshot 2024-09-27 at 18 20 25

File: features/admin.core.v1/models/common.ts [hunk 15]
```diff
@@ -1,10 +1,19 @@
 /**
- * Copyright (c) 2023, WSO2 LLC. (https://www.wso2.com). All Rights Reserved.
+ * Copyright (c) 2024, WSO2 LLC. (https://www.wso2.com).
```
Comment (pavinduLakshan) [inline features/admin.core.v1/models/common.ts:2]:
```suggestion
 * Copyright (c) 2023-2024, WSO2 LLC. (https://www.wso2.com).
```

File: modules/react-components/src/components/page-header/page-header.tsx [hunk 16]
```diff
@@ -301,6 +306,9 @@ export const PageHeader: React.FunctionComponent<PageHeaderPropsInterface> = (
                                 )
                         )
                     }
+                    {
+                        alertBanner
+                    }
```
Comment (pavinduLakshan) [inline modules/react-components/src/components/page-header/page-header.tsx:311]:
```suggestion
                    { alertBanner }
```

File: modules/theme/src/themes/default/elements/label.overrides [hunk 17]
```diff
@@ -142,6 +142,17 @@
...
 }
 
+/*******************************
+      Application outdated Label
+*******************************/
+
+.ui.label {
+    &.application-outdated-label {
+        color: #fd9200;
+        background-color: #fff4e5;
```
Comment (pavinduLakshan) [inline modules/theme/src/themes/default/elements/label.overrides:152]: Can we use pre defined css variables for this?

cc: @savindi7

File: modules/i18n/src/translations/en-US/portals/applications.ts [hunk 18]
```diff
@@ -1643,6 +1643,37 @@ export const applications: ApplicationsNS = {
...
+                    },
+                    fields: {
+                        commonInstruction: "Following behavioral changes will be applied upon update.",
+                        versions: {
+                            version100: {
+                                removeUsernameFromIntrospectionRespForAppTokens: {
+                                    instruction: "Introspection response for application access token will not include the username attribute."
+                                },
+                                useClientIdAsSubClaimOfAppTokens: {
+                                    instruction: "Application access token sub attribute will be client_id generated for an application."
```
Comment (pavinduLakshan) [inline modules/i18n/src/translations/en-US/portals/applications.ts:1671]:
```suggestion
                                    instruction: "The sub attribute of the application access token will be the client_id generated for the application."
```

File: modules/i18n/src/translations/en-US/portals/applications.ts [hunk 19]
```diff
@@ -1643,6 +1643,37 @@ export const applications: ApplicationsNS = {
...
+                        message: "Proceeding the action without making relevant change will cause the client application behavior break.",
+                        content: "This action is irreversible and will permanently update the application.",
+                        assertionHint: "Please confirm your action"
+                    },
+                    fields: {
+                        commonInstruction: "Following behavioral changes will be applied upon update.",
+                        versions: {
+                            version100: {
+                                removeUsernameFromIntrospectionRespForAppTokens: {
+                                    instruction: "Introspection response for application access token will not include the username attribute."
```
Comment (pavinduLakshan) [inline modules/i18n/src/translations/en-US/portals/applications.ts:1668]:
```suggestion
                                    instruction: "The introspection response for an application access token will not include the username attribute."
```

File: modules/i18n/src/translations/en-US/portals/applications.ts [hunk 20]
```diff
@@ -1643,6 +1643,37 @@ export const applications: ApplicationsNS = {
...
+                        viewButton: "View Details",
+                        hideButton: "Hide Details",
+                        cancelButton: "Ignore Once"
+                    },
+                    label: "Outdated",
+                    documentationHint: "More Details",
+                    confirmationModal: {
+                        header: "Have you done the relevant changes?",
+                        message: "Proceeding the action without making relevant change will cause the client application behavior break.",
+                        content: "This action is irreversible and will permanently update the application.",
```
Comment (pavinduLakshan) [inline modules/i18n/src/translations/en-US/portals/applications.ts:1660]:
```suggestion
                        content: "This action is irreversible and will result in a permanent update to the application.",
```

File: modules/i18n/src/translations/en-US/portals/applications.ts [hunk 21]
```diff
@@ -1643,6 +1643,37 @@ export const applications: ApplicationsNS = {
...
+                        content: "This application is using an outdated behavior of applications.",
+                        viewButton: "View Details",
+                        hideButton: "Hide Details",
+                        cancelButton: "Ignore Once"
+                    },
+                    label: "Outdated",
+                    documentationHint: "More Details",
+                    confirmationModal: {
+                        header: "Have you done the relevant changes?",
+                        message: "Proceeding the action without making relevant change will cause the client application behavior break.",
```
Comment (pavinduLakshan) [inline modules/i18n/src/translations/en-US/portals/applications.ts:1659]:
```suggestion
                        message: "Proceeding without making the necessary changes will cause the client application's behavior to break.",
```

**Review by pavinduLakshan — COMMENTED** [review]
shall we get the UI text reviewed from the doc team?

cc: @himeshsiriwardana

File: features/admin.applications.v1/constants/application-management.ts [hunk 22]
```diff
@@ -41,6 +41,11 @@ export class ApplicationManagementConstants {
     public static readonly SYSTEM_APPS: string[] = [ this.CONSOLE_APP_NAME ];
     public static readonly DEFAULT_APPS: string[] = [ this.MY_ACCOUNT_APP_NAME ];
 
+    /**
+     * Application latest version.
+     */
+    public static readonly LATEST_VERSION: string = "v1.0.0";
```
Comment (pavinduLakshan) [inline features/admin.applications.v1/constants/application-management.ts:47]: Is hardcoding this here intentional?

File: features/admin.applications.v1/utils/application-management-utils.ts [hunk 23]
```diff
@@ -104,6 +104,28 @@ export class ApplicationManagementUtils {
             });
     }
 
+    public static getIfAppIsOutdated(applicationVersion: string, grantTypes?: string[]): boolean {
```
Comment (pavinduLakshan) [inline features/admin.applications.v1/utils/application-management-utils.ts:107]:
```suggestion
    public static isApplicationOutdated(applicationVersion: string, grantTypes?: string[]): boolean {
```

File: features/admin.applications.v1/constants/application-management.ts (hunk 22 above)
Comment (pavinduLakshan) [reply features/admin.applications.v1/constants/application-management.ts:47]: Let's add a comment

File: features/admin.applications.v1/utils/application-management-utils.ts (hunk 23 above)
Comment (mpmadhavig) [reply features/admin.applications.v1/utils/application-management-utils.ts:107]: 0bd8ceec802ce087b6d7e10d520ad8d787c6e5f1

File: features/admin.applications.v1/constants/application-management.ts (hunk 22 above)
Comment (mpmadhavig) [reply features/admin.applications.v1/constants/application-management.ts:47]: 0bd8ceec802ce087b6d7e10d520ad8d787c6e5f1

**Comment by mpmadhavig** [comment]
Due to the comment in https://github.com/wso2-enterprise/iam-product-management/issues/295#issuecomment-2385261432, the label will not be displayed.

File: features/admin.applications.v1/utils/application-management-utils.ts [hunk 24]
```diff
@@ -104,6 +104,28 @@ export class ApplicationManagementUtils {
             });
     }
 
+    public static isApplicationOutdated(applicationVersion: string, grantTypes?: string[]): boolean {
+
+        if (applicationVersion != undefined
+                && grantTypes.includes(ApplicationManagementConstants.CLIENT_CREDENTIALS_GRANT)) {
```
Comment (pavinduLakshan) [inline features/admin.applications.v1/utils/application-management-utils.ts:110]:
```suggestion
        if (
             applicationVersion != undefined
             && grantTypes.includes(ApplicationManagementConstants.CLIENT_CREDENTIALS_GRANT)
        ) {
```

**Review by pavinduLakshan — DISMISSED** [review]

**Review by pavinduLakshan — APPROVED** [review]

**Comment by himeshsiriwardana** [comment]
**Application Update Required.**

We have updated the behavior of application access tokens as follows::

- The `sub` attribute of an application access token now returns the `client_id` generated for the application, instead of the `userid` of the application owner.
- The introspection responses for application access tokens no longer return the `username` attribute.

You may click the button below to update the application’s behavior. Once updated, be sure to update any usages of these attributes accordingly. [Learn more]

**Comment by mpmadhavig** [comment]
IS banner view:
Screenshot 2024-10-08 at 15 16 10

**Review by pavinduLakshan — APPROVED** [review]

File: features/admin.applications.v1/pages/application-edit.tsx [hunk 25]
```diff
@@ -129,6 +149,25 @@ const ApplicationEditPage: FunctionComponent<ApplicationEditPageInterface> = (
...
+    } = useGetApplicationInboundConfigs(applicationId, SupportedAuthProtocolName.OIDC, !!applicationId);
+
+    const [ viewBannerDetails, setViewBannerDetails ] = useState<boolean>(false);
+    const [ displayBanner, setDisplayBanner ] = useState<boolean>(false);
+    const [ bannerUpdateLoading, setBannerUpdateLoading ] = useState<boolean>(false);
+    const [ showConfirmationModal, setShowConfirmationModal ] = useState<boolean>(false);
+    const [ formData, setFormdata ] = useState<ApplicationInterface>(undefined);
+
+    useEffect(() => {
+        if (application != undefined && applicationInboundConfigs != undefined) {
```
Comment (savindi7) [inline features/admin.applications.v1/pages/application-edit.tsx:163]:
```suggestion
        if (application && applicationInboundConfigs) {
```

File: features/admin.applications.v1/pages/application-edit.tsx [hunk 26]
```diff
@@ -506,6 +545,297 @@ const ApplicationEditPage: FunctionComponent<ApplicationEditPageInterface> = (
...
+        return (
+            displayBanner &&
+                (
+                    <div className="banner-wrapper">
+                        <Alert
+                            className={ classes }
+                            severity="warning"
+                            action={
+                                (
+                                    <Box display="flex">
```
Comment (savindi7) [inline features/admin.applications.v1/pages/application-edit.tsx:565]: let's not use inline styles and add these in the stylesheets.

File: features/admin.applications.v1/pages/application-edit.tsx [hunk 27]
```diff
@@ -506,6 +545,297 @@ const ApplicationEditPage: FunctionComponent<ApplicationEditPageInterface> = (
...
+                                                    t("applications:forms.inboundOIDC.sections"
+                                                            + ".outdatedApplications.alert.hideButton")
+                                            }
+                                        </Button>
+                                        <Button
+                                            data-componentid={ `${componentId}-outdated-app-ignore-button` }
+                                            className="ignore-once-button"
+                                            onClick={ () => setDisplayBanner(false) }>
+                                            <Icon
+                                                link={ true }
```
Comment (savindi7) [inline features/admin.applications.v1/pages/application-edit.tsx:583]:
```suggestion
                                                link
```

File: features/admin.applications.v1/pages/application-edit.tsx [hunk 28]
```diff
@@ -506,6 +545,297 @@ const ApplicationEditPage: FunctionComponent<ApplicationEditPageInterface> = (
...
+                                                onClick={ () => setDisplayBanner(false) }
+                                                className=""
```
Comment (savindi7) [inline features/admin.applications.v1/pages/application-edit.tsx:585]:
```suggestion
```

File: features/admin.applications.v1/pages/application-edit.tsx [hunk 29]
```diff
@@ -33,21 +42,29 @@ import { ExtensionTemplateListInterface } from "@wso2is/admin.template-core.v1/m
...
+import classNames from "classnames";
 import cloneDeep from "lodash-es/cloneDeep";
 import React, { FunctionComponent, ReactElement, useEffect, useMemo, useRef, useState } from "react";
-import { useTranslation } from "react-i18next";
+import { Trans, useTranslation } from "react-i18next";
 import { useDispatch, useSelector } from "react-redux";
 import { RouteComponentProps } from "react-router";
 import { Dispatch } from "redux";
-import { Label } from "semantic-ui-react";
+import { Icon, Label } from "semantic-ui-react";
```
Comment (savindi7) [inline features/admin.applications.v1/pages/application-edit.tsx:64]: Let's use the oxygen components here

File: modules/theme/src/themes/default/elements/label.overrides [hunk 30]
```diff
@@ -142,6 +142,17 @@
     }
 }
 
+/*******************************
+      Application outdated Label
+*******************************/
+
+.ui.label {
+    &.application-outdated-label {
```
Comment (savindi7) [inline modules/theme/src/themes/default/elements/label.overrides:150]: Are we using these styles anywhere?

**Review by pavinduLakshan — APPROVED** [review]

**Comment by mpmadhavig** [comment]
Will make the necessary changes via the next phase of this effort.

**Pull request merged**